docs/_getting-started/dev-mode.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Dev mode
+---
+
+Geth has a development mode which sets up a single node Ethereum test network with a number of options optimized for developing on local machines. You enable it with the `--dev` argument.
+
+Starting geth in dev mode does the following:
+
+-   Initializes the data directory with a testing genesis block
+-   Sets max peers to 0
+-   Turns off discovery by other nodes
+-   Sets the gas price to 0
+-   Uses the ethash consensus engine with clique headers to allows blocks to be mined as fast as your CPU and memory can handle
+-   Prevents the consensus (proof of work) difficulty from ever increasing
+-   Permits 0 block times in the clique consensus engine and forbid empty blocks for 0 periods
```

Comment:
0 block times - not sure what that means. It's possible to set `dev.period` to have e.g. one block every second, but by default it mines if there are any transactions waiting to be included.

Reply:
I can't quite remember where this information came from, and I don't even understand it myself anymore now.

I think it was something to do with speeding up testing by effectively disabling features of the consensys engine. Is it best to cut this line or paraphrase what you say above?

Reply:
Write "Uses on-demand block generation, producing blocks when there are transactions waiting to be mined"